test/lib/ufe/testLight.py

```diff
@@ -0,0 +1,173 @@
+
+        usdAttr.Set(0.5)
+        self.assertAlmostEqual(0.5, ufeLight.diffuse())
+
+        ufeLight.diffuse(0.9)
+        self.assertAlmostEqual(usdAttr.Get(), 0.9)
+
+    def _TestSpecular(self, ufeLight, usdLight):
+        usdAttr = usdLight.GetAttribute('inputs:specular')
+        self.assertAlmostEqual(usdAttr.Get(), ufeLight.specular())
+
+        usdAttr.Set(0.5)
+        self.assertAlmostEqual(0.5, ufeLight.specular())
+
+        ufeLight.specular(0.9)
+        self.assertAlmostEqual(usdAttr.Get(), 0.9)
+
+    def _TestShadowEnable(self, ufeLight, usdLight):
+        usdAttr = usdLight.GetAttribute('inputs:shadow:enable')
+        self.assertEqual(usdAttr.Get(), ufeLight.shadowEnable())
+
+        usdAttr.Set(False)
+        self.assertEqual(False, ufeLight.shadowEnable())
+
+        ufeLight.shadowEnable(True)
+        self.assertAlmostEqual(usdAttr.Get(), True)        
+
+    def _TestColor(self, ufeLight, usdLight):
+        usdAttr = usdLight.GetAttribute('inputs:color')
+        self.assertAlmostEqual(usdAttr.Get()[0], ufeLight.color().r())
+        self.assertAlmostEqual(usdAttr.Get()[1], ufeLight.color().g())
+        self.assertAlmostEqual(usdAttr.Get()[2], ufeLight.color().b())
+
+    def _TestShadowColor(self, ufeLight, usdLight):
+        usdAttr = usdLight.GetAttribute('inputs:shadow:color')
+        self.assertAlmostEqual(usdAttr.Get()[0], ufeLight.shadowColor().r())
+        self.assertAlmostEqual(usdAttr.Get()[1], ufeLight.shadowColor().g())
+        self.assertAlmostEqual(usdAttr.Get()[2], ufeLight.shadowColor().b())
+
+    def _TestSphereProps(self, ufeLight, usdLight):
+        usdAttr = usdLight.GetAttribute('inputs:radius')
+        self.assertAlmostEqual(usdAttr.Get(), ufeLight.sphereInterface().sphereProps().radius)
+        self.assertEqual(usdAttr.Get() == 0, ufeLight.sphereInterface().sphereProps().asPoint)
+
+    def _TestConeProps(self, ufeLight, usdLight):
+        usdAttrFocus = usdLight.GetAttribute('inputs:shaping:focus')
+        usdAttrAngle = usdLight.GetAttribute('inputs:shaping:cone:angle')
+        usdAttrSoftness = usdLight.GetAttribute('inputs:shaping:cone:softness')
+        self.assertAlmostEqual(usdAttrFocus.Get(), ufeLight.coneInterface().coneProps().focus)
+        self.assertAlmostEqual(usdAttrAngle.Get(), ufeLight.coneInterface().coneProps().angle)
+        self.assertAlmostEqual(usdAttrSoftness.Get(), ufeLight.coneInterface().coneProps().softness)
+
+    def testUsdLight(self):
+        self._StartTest('SimpleLight')
+        mayaPathSegment = mayaUtils.createUfePathSegment('|stage|stageShape')
+        lightUsdPathSegment = usdUtils.createUfePathSegment('/lights/simpleLight')
+        lightPath = ufe.Path([mayaPathSegment, lightUsdPathSegment])
+        lightItem = ufe.Hierarchy.createItem(lightPath)
+
+        ufeLight = ufe.Light.light(lightItem)
+        usdLight = usdUtils.getPrimFromSceneItem(lightItem)
+        self.assertEqual(ufeLight.type(), ufe.Light.Spot)
```

Comment:
We need to test all the different light types we support. We should test setting values that don't exist on a given light type, expecting failure.